### PR TITLE
Adding depthai_filters package in Humble

### PR DESCRIPTION
**Pull request by Serafadam**
* Segmentation overlay
* Detection 2D overlay
* WLS filter
* Fixes bug in https://github.com/luxonis/depthai-ros/issues/248

### CODE REVIEW

**Comment by saching13** [comment]
Will look at it after https://github.com/luxonis/depthai-ros/pull/246 is merged since it is built on-top of that.

**Comment by Serafadam** [comment]
@saching13 #246 is merged. Changes in `depthai_ros_driver` outside of bugfix are cosmetic and related to parameter getters, api has been simplified here to make code more readable.

File: depthai_filters/include/depthai_filters/detection2d_overlay.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    void overlayCB(const sensor_msgs::msg::Image::ConstSharedPtr& preview, const vision_msgs::msg::Detection2DArray::ConstSharedPtr& detections);
+
+    message_filters::Subscriber<sensor_msgs::msg::Image> previewSub;
+    message_filters::Subscriber<vision_msgs::msg::Detection2DArray> detSub;
+
+    typedef message_filters::sync_policies::ApproximateTime<sensor_msgs::msg::Image, vision_msgs::msg::Detection2DArray> syncPolicy;
+    std::unique_ptr<message_filters::Synchronizer<syncPolicy>> sync;
+    rclcpp::Publisher<sensor_msgs::msg::Image>::SharedPtr overlayPub;
+    const std::vector<std::string> labelMap = {"background", "aeroplane", "bicycle",     "bird",  "boat",        "bottle", "bus",
```
Comment (saching13) [inline depthai_filters/include/depthai_filters/detection2d_overlay.hpp:25]: can we add option to overload this map ? Incase model was retrained with different labels ?

File: depthai_filters/include/depthai_filters/segmentation_overlay.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    void onInit();
+
+    void overlayCB(const sensor_msgs::msg::Image::ConstSharedPtr& preview, const sensor_msgs::msg::Image::ConstSharedPtr& segmentation);
+
+    message_filters::Subscriber<sensor_msgs::msg::Image> previewSub, segSub;
+
+    typedef message_filters::sync_policies::ApproximateTime<sensor_msgs::msg::Image, sensor_msgs::msg::Image> syncPolicy;
+    std::unique_ptr<message_filters::Synchronizer<syncPolicy>> sync;
+    rclcpp::Publisher<sensor_msgs::msg::Image>::SharedPtr overlayPub;
+    const std::vector<std::string> labelMap = {"background", "aeroplane", "bicycle",     "bird",  "boat",        "bottle", "bus",
```
Comment (saching13) [inline depthai_filters/include/depthai_filters/segmentation_overlay.hpp:23]: can we add option to overload this map ? Incase model was retrained with different labels ?

File: depthai_filters/include/depthai_filters/segmentation_overlay.hpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    explicit SegmentationOverlay(const rclcpp::NodeOptions& options);
+    void onInit();
+
+    void overlayCB(const sensor_msgs::msg::Image::ConstSharedPtr& preview, const sensor_msgs::msg::Image::ConstSharedPtr& segmentation);
+
+    message_filters::Subscriber<sensor_msgs::msg::Image> previewSub, segSub;
+
+    typedef message_filters::sync_policies::ApproximateTime<sensor_msgs::msg::Image, sensor_msgs::msg::Image> syncPolicy;
+    std::unique_ptr<message_filters::Synchronizer<syncPolicy>> sync;
+    rclcpp::Publisher<sensor_msgs::msg::Image>::SharedPtr overlayPub;
```
Comment (saching13) [inline depthai_filters/include/depthai_filters/segmentation_overlay.hpp:22]: why don't we just use [visualization_msgs](http://docs.ros.org/en/noetic/api/visualization_msgs/html/index-msg.html) for this ? that way rviz or foxglove can use it to see without having to republish a different image. We can keep both options depending on they type of user and where they are running. 
if running on a turtlebot. it makes sense to have it as `visualization_msgs`

**Review by saching13 — COMMENTED** [review]
overall LGTM
Added one additional change w.r.t visualization_msgs

File: depthai_filters/include/depthai_filters/segmentation_overlay.hpp (hunk 2 above)
Comment (Serafadam) [reply depthai_filters/include/depthai_filters/segmentation_overlay.hpp:23]: Sure, I'll add it as a parameter

File: depthai_filters/include/depthai_filters/segmentation_overlay.hpp (hunk 3 above)
Comment (Serafadam) [reply depthai_filters/include/depthai_filters/segmentation_overlay.hpp:22]: Sounds good for detection, I'm not sure about segmentation as it's more of an example showing usage

File: depthai_filters/include/depthai_filters/segmentation_overlay.hpp (hunk 2 above)
Comment (Serafadam) [reply depthai_filters/include/depthai_filters/segmentation_overlay.hpp:23]: On the other hand, Vision msgs have custom messages for information about labels:
https://github.com/ros-perception/vision_msgs/blob/ros2/vision_msgs/msg/LabelInfo.msg (not available in Noetic)

https://github.com/ros-perception/vision_msgs/blob/ros2/vision_msgs/msg/VisionInfo.msg

I think it's a better approach here, though I'd add that in a separate PR before next release

**Pull request merged**